### ISSUE TITLE
Scalable workflow pour les extractions

### ISSUE THREAD

**Issue by ThomasSoeiro** [SNDStoolers/sndsTools#15]
Salut,

Je pense que ça va coincer pour les grosses extractions :
https://github.com/strayMat/paresnds/blob/a0551eb889637f836c011bf8b87aec13f97cbf21/R/retrieve_atc.R#L88-L94

- Dans le cas de l'import dans un _data frame_ dans R, il est probable que ce soit très long et/ou se solde par un célèbre `Erreur : impossible d'allouer un vecteur de taille n Go`.
- Dans le cas de l'enregistrement en `.RDS`, il est probable que l'espace projet soit trop petit.

Je pense qu'il vaut mieux garder la table dans Oracle, faire le _data management_ avec Oracle, puis importer seulement la table une fois qu'elle est propre (et donc on espère plus petite).

**Comment by abelloir-zz** [comment]
Dans les mises à jour que je suis en train de faire sur l'ensemble des fonctions d'extraction, je supprime cette option de sauvegarder la table de sortie au sein du script d'extraction : à la place, je renvoie la table en output de fonction.

Mais effectivement le problème de la taille limite des tables demeure. Il y a toujours un moment où les données finissent par être rapatriées dans R. Plus tôt on les sort plus il y a en effet un risque de dépasser la mémoire disponible.

Je pense que dans la plupart des cas, si l'on ne travaille pas sur la population nationale entière, ce problème ne sera rencontré qu'avec les délivrances de médicaments qui sont de loin les tables les plus volumineuses. J'avais initialement dans la fonction d'extraction  des délivrances de médicaments un argument `database_output_table_name` (nom sous lequel enregistrer la table de sortie sous Oracle). Je l'avais enlevé car dans le cadre de mon projet, car les données de sortie brutes restaient d'une taille raisonnable, et par soucis d'uniformité avec les autres fonctions d'extraction.

Pour la fonction  d'extraction  des délivrances de médicaments, je peux rétablir cet argument, et je peux rajouter en même temps un paramètre `do_not_collect` ou équivalent pour éviter le renvoi de la table et le chargement en mémoire sur R

**Comment by ThomasSoeiro** [comment]
> Je pense que dans la plupart des cas, si l'on ne travaille pas sur la population nationale entière, ce problème ne sera rencontré qu'avec les délivrances de médicaments qui sont de loin les tables les plus volumineuses.

Je suis d'accord, mais ça me semble dommage de designer le paquet sans prendre en compte les utilisateurs qui travaillent sur la population nationale. Avec la population nationale, j'ai eu ce problème également avec le PMSI MCO (tables B + UM + D sans _reshape wide_).

Je propose de créer les tables dans Oracle et laisser l'utilisateur choisir ce qu'il veut en faire. Il me semble que la complexification pour l'utilisateur est minime. Pour importer la table, il suffit d'un `df_r <- dbReadTable(con, "TABLE_ORACLE")` (ou équivalent en `{dbplyr}`).

**Comment by abelloir-zz** [comment]
Je rajoute à toutes les fonctions d'axtarction cette option de laisser la table sous Oracle sans la sortir, via un paramètre `output_table_name`. Checklist :
- [ ] R/extract_long_term_disease.R,mergé
- [ ]  R/extract_drug_dispenses.R, reste à merger
- [ ]  R/retrieve_consultations.R
- [ ]  R/extract_hospital_consultations.R
- [ ]  R/extract_hospital_stays.R
